xcube/core/store/fs/store.py

```diff
@@ -556,14 +580,16 @@ def _guess_accessor_id_parts(self, data_id: str, require=True) \
             -> Optional[Tuple[str, str, str]]:
         assert_given(data_id, 'data_id')
         ext = self._get_filename_ext(data_id)
-        data_type_alias = _FILENAME_EXT_TO_DATA_TYPE_ALIAS.get(ext)
-        format_name = _FILENAME_EXT_TO_FORMAT.get(ext)
-        if data_type_alias is None or format is None:
+        data_type_aliases = None
+        format_id = _FILENAME_EXT_TO_FORMAT.get(ext)
+        if format_id is not None:
+            data_type_aliases = _FORMAT_TO_DATA_TYPE_ALIASES.get(format_id)
+        if data_type_aliases is None or format_id is None:
             if require:
                 raise DataStoreError(f'Cannot determine data type for '
                                      f' data resource {data_id!r}')
             return None
-        return data_type_alias, format_name, self.protocol
+        return data_type_aliases[0], format_id, self.protocol
```

Comment:
This is the part where we arbitrarily pick one data_type_alias (the first one). Depending on how this method has been used, it might introduce problems if we omit other data type aliases.

Reply:
No, the implementation defines the first alias to be the one. This is stored in constants defined at the top of the file. Secondly, they are just aliases, so they all point to the same data type.

Reply:
...and so far, tests are green too.

Reply:
They stop running green if we change the order of aliases in _FORMAT_TO_DATA_TYPE_ALIASES. However, I guess it is okay if we say that the first data type is the primary one,